### PR TITLE
 fs: add TestCopyReflinkWithXFS

### PR DESCRIPTION
**Pull request by AkihiroSuda**
This PR makes sure `CopyFile()` supports reflink on XFS.

This PR also covers https://github.com/containerd/continuity/pull/127

### CODE REVIEW

File: fs/copy_linux_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		loopback.Close()
+	}()
+
+	aPath := filepath.Join(mnt, "a")
+	aSize := int64(100 << 20) // 100MB
+	a, err := os.Create(aPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := io.CopyN(a, rand.Reader, aSize); err != nil {
```
Comment (dmcgowan) [inline fs/copy_linux_test.go]: Don't use crypto rand for this

File: testutil/loopback_linux.go [hunk 2]
```diff
@@ -49,5 +69,47 @@ func NewLoopback(size int64) (string, func() error, error) {
...
+// SoftSize returns st_size
+func (l *Loopback) SoftSize() (int64, error) {
+	st, err := os.Stat(l.File)
+	if err != nil {
+		return 0, err
+	}
+	return st.Size(), nil
+}
+
+// HardSize returns st_blocks * 512
```
Comment (dmcgowan) [inline testutil/loopback_linux.go]: Can you link to somewhere that justifies 512 as a hardcoded value

File: fs/copy_linux_test.go (hunk 1 above)
Comment (AkihiroSuda) [reply fs/copy_linux_test.go]: What should I do instead?

File: testutil/loopback_linux.go (hunk 2 above)
Comment (AkihiroSuda) [reply testutil/loopback_linux.go]: http://man7.org/linux/man-pages/man2/fstat.2.html

> st_blocks
>  This field indicates the number of blocks allocated to the file, in 512-byte units.

File: fs/copy_linux_test.go (hunk 1 above)
Comment (dmcgowan) [reply fs/copy_linux_test.go]: `math/rand` should work just fine here and be much faster

File: fs/copy_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
+// +build linux
```
Comment (dmcgowan) [inline fs/copy_linux_test.go]: The file header checking was added, that needs to be added here to pass CI

**Comment by dmcgowan** [comment]
LGTM after header added to make CI hapy

**Comment by AkihiroSuda** [comment]
updated

**Pull request merged**